content/storage/furps/frontends-filesharing.md

```diff
@@ -0,0 +1,35 @@
+# Serving Frontends, Modules, and Files
+
+## Functionality
+
+1. **File sharing.** Any user in the network should be able to share files (frontend bundles and modules) such that those can be downloaded by any other user in the network.
+1. **Content addressability.** Files should be addressable by a content identifier, such that users are sure that the content they download is the content they expect.
+1. **Organic replication.** Users that download a file should be able to provide it to other users in the network.
+1. **Caching and storing.** Nodes should allow files to be either _cached_ or _stored_. Caching is constrained by a quota, and any file placed onto a node node should be cached unless explicitly told otherwise. Cached files are _evicted_ once the node runs out of quota. Optionally, a file might be marked as _stored_, at which point it no longer occupies the caching quota, and is simply preserved on disk.
```

Comment:
```suggestion
1. **Caching and storing.** Nodes should allow files to be either _cached_ or _stored_. Caching is constrained by a quota, and any file placed onto a node should be cached unless explicitly told otherwise. Cached files are _evicted_ once the node runs out of caching quota in a LRU manner. Optionally, a file might be explicitly marked as _stored_, at which point it no longer occupies the caching quota, and is simply preserved on disk. Storage space is limited by a separate storage quota.
```